### PR TITLE
fix: Improve calculation orchestration error handling

### PR DESCRIPTION
**Pull request by ebbeknudsen**
# Description

Improve calculation orchestration error handling
- Handle exception when raising orchestration event to log a warning instead of throwing exception
- Handle ActorMessagesEnqueued timeout in orchestration to set failed status instead of throwing exception

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [x] Subsystem tests have been tested (by manually deploying to `dev_002`) => https://github.com/Energinet-DataHub/dh3-environments/actions/runs/9500653392

### CODE REVIEW

File: source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/WholesaleInboxTriggerTests.cs [hunk 1]
```diff
@@ -142,17 +143,10 @@ public async Task GivenWholesaleServicesRequest_WhenEventIsHandled_FunctionCompl
     public async Task GivenActorMessagesEnqueued_WhenEventIsHandled_FunctionCompletes()
```
Comment (MadsDue) [inline source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/WholesaleInboxTriggerTests.cs:143]: If the new assert is important for this test, I would believe it should be part of the method name.

File: source/dotnet/wholesale-api/Orchestrations/Functions/WholesaleInbox/ActorMessagesEnqueuedV1RequestHandler.cs [hunk 2]
```diff
@@ -45,10 +46,27 @@ public Task ProcessAsync(ServiceBusReceivedMessage receivedMessage, string refer
...
+                    cancellationToken)
+                .ConfigureAwait(false);
+        }
+        catch (RpcException exception)
+        {
+            // Exception thrown if OrchestrationInstanceId couldn't be found:
+            //      Grpc.Core.RpcException: Status(StatusCode="Unknown", Detail="Exception was thrown by handler.")
+            // This can happen if the orchestration has finished and a duplicate ActorMessagesEnqueued event is received
+            // or if the orchestration has timed out and the ActorMessagesEnqueued is received afterwards
+            _logger.LogWarning(
```
Comment (MadsDue) [inline source/dotnet/wholesale-api/Orchestrations/Functions/WholesaleInbox/ActorMessagesEnqueuedV1RequestHandler.cs:64]: Is this also a warning if it happens in produktion?

Comment (ebbeknudsen) [reply source/dotnet/wholesale-api/Orchestrations/Functions/WholesaleInbox/ActorMessagesEnqueuedV1RequestHandler.cs:64]: Yes - it has been decided that our way of handling idempotency when receiving the event is that we shouldn't throw an exception in this case, only log a warning.

File: source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/WholesaleInboxTriggerTests.cs (hunk 1 above)
Comment (ebbeknudsen) [reply source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/WholesaleInboxTriggerTests.cs]: I've changed the name and added a comment 👍

File: source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/WholesaleInboxTriggerTests.cs [hunk 3]
```diff
@@ -161,6 +155,10 @@ public async Task GivenActorMessagesEnqueued_WhenEventIsHandled_FunctionComplete
         // Handling a MessagesEnqueuedV1 should raise an event to the Durable Task client, which we cannot test without
         // using some kind of mock, but we can atleast verify that the function completes.
         await AssertWholesaleInboxTriggerIsCompleted();
+        var functionHostLogs = Fixture.AppHostManager.GetHostLogSnapshot();
+        functionHostLogs.Should()
```
[the remark this answers is not in the document]
Comment (ebbeknudsen) [reply source/dotnet/wholesale-api/Orchestrations.IntegrationTests/Functions/WholesaleInboxTriggerTests.cs:161]: I don't understand this question - lets have a talk :)

**Review by MadsDue — APPROVED** [review]

**Pull request merged**